src/main/java/hudson/plugins/ec2/SpotConfiguration.java

```diff
@@ -1,15 +1,20 @@
 package hudson.plugins.ec2;
 
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
 import java.util.Objects;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
-public final class SpotConfiguration {
+public final class SpotConfiguration extends AbstractDescribableImpl<SpotConfiguration>  {
     public final boolean useBidPrice;
-    public final String spotMaxBidPrice;
-    public final boolean fallbackToOndemand;
-    public final int spotBlockReservationDuration;
+    public String spotMaxBidPrice;
```

Comment:
I dislike this change as this is now public and modifiable

Reply:
It doesn't appear to be used outside this plugin so you could probably make it private https://github.com/search?q=org%3Ajenkinsci+spotMaxBidPrice&type=Code

Reply:
I making them private, should we add a note or something in the changelog to say it may break compatibility?